packages/actor-init-sparql/lib/ActorInitSparql.ts

```diff
@@ -25,57 +25,128 @@ export {
+
+  public static async buildContext(args: minimist.ParsedArgs, queryOrHttp: number, helpMessage: string,
+    queryString?: string): Promise<any> {
+    const query: boolean = queryOrHttp === 0;
+
+    // Print version information
+    if (args.v || args.version) {
+      const comunicaVersion: string = require('../package.json').version;
+      const dev: string = this.isDevelopmentEnvironment() ? '(dev)' : '';
+      const nodeVersion: string = process.version;
+      const npmVersion: string = await this.getScriptOutput('npm -v', '_NPM is unavailable_');
+      const yarnVersion: string = await this.getScriptOutput('yarn -v', '_Yarn is unavailable_');
+      const os = `${OS.platform()} (${OS.type()} ${OS.release()})`;
+
+      const message = require('streamify-string')(`| software            | version
 | ------------------- | -------
 | Comunica Init Actor | ${comunicaVersion} ${dev}
 | node                | ${nodeVersion}
 | npm                 | ${npmVersion}
 | yarn                | ${yarnVersion}
 | Operating System    | ${os}
-`) };
-      }
+`);
+
+      return { stderr: message };
+    }
 
+    if (args.h || args.help ||
+        (query && (!args.listformats && (!queryString && (!(args.q || args.f) && args._.length < (args.c ? 1 : 2) ||
+        args._.length < (args.c ? 0 : 1))))) ||
+        (!query && ((args.c && args._.length > 0) || (!args.c && args._.length === 0)))) {
       // Print command usage
-      return { stderr: require('streamify-string')(`comunica-sparql evaluates SPARQL queries
-
-Usage:
-  comunica-sparql http://fragments.dbpedia.org/2016-04/en [-q] 'SELECT * WHERE { ?s ?p ?o }'
-  comunica-sparql http://fragments.dbpedia.org/2016-04/en [-q] '{ hero { name friends { name } } }' -i graphql
-  comunica-sparql http://fragments.dbpedia.org/2016-04/en [-f] query.sparql'
-  comunica-sparql http://fragments.dbpedia.org/2016-04/en https://query.wikidata.org/sparql ...
-  comunica-sparql hypermedia@http://fragments.dbpedia.org/2016-04/en sparql@https://query.wikidata.org/sparql ...
-
-Options:
-  -q            evaluate the given SPARQL query string
-  -f            evaluate the SPARQL query in the given file
-  -c            use the given JSON configuration file (e.g., config.json)
-  -t            the MIME type of the output (e.g., application/json)
-  -i            the query input format (e.g., graphql, defaults to sparql)
-  -b            base IRI for the query (e.g., http://example.org/)
-  -l            sets the log level (e.g., debug, info, warn, ... defaults to warn)
-  -d            sets a datetime for querying Memento-enabled archives
-  -p            delegates all HTTP traffic through the given proxy (e.g. http://myproxy.org/?uri=)
-  --lenient     if failing requests and parsing errors should be logged instead of causing a hard crash
-  --help        print this help message
-  --listformats prints the supported MIME types
-  --version     prints version information
-`) };
+      return { stderr: helpMessage };
```

Comment:
Instead of returning this as the context, I would throw an error here (reject the promise). Then we can put a try-catch around this when we call this function, and we can handle it accordingly.